### PR TITLE
suggested error handling approach

### PR DESCRIPTION
**Pull request by sandheep45**

### CODE REVIEW

**Comment by sandheep45** [comment]
I didn't got the point for `apiManager.ts` @prasenjeet-symon

**Comment by prasenjeet-symon** [comment]
> I didn't got the point for `apiManager.ts` @prasenjeet-symon

Its main task is to listen for all the network error and the hook information as well ( because we are using react-query API request goes through the react-query hook ). In the future, it will communicate with the server about the error ( error reporting  ). From here we can push the error reporting part out of the main thread.

File: packages/client/src/api/api.ts [hunk 1]
```diff
@@ -13,12 +13,24 @@ import AxiosClient from "./apiClient";
...
+      password,
+    });
 
-  return axiosClient;
+    return axiosClient;
+  } catch (error) {
+    if (error instanceof AxiosError) {
+      const { response } = error;
+
+      throw {
```
Comment (prasenjeet-symon) [inline packages/client/src/api/api.ts:27]: We can simply throw the error

File: packages/client/src/api/api.ts [hunk 2]
```diff
@@ -13,12 +13,24 @@ import AxiosClient from "./apiClient";
...
+        message: response?.data.error,
+      };
+    }
+
+    throw new Error(error as string);
```
Comment (prasenjeet-symon) [inline packages/client/src/api/api.ts:32]: We can simply throw the incoming error

File: packages/client/src/hooks/authentication.hooks.ts
Comment (prasenjeet-symon) [inline packages/client/src/hooks/authentication.hooks.ts:1]: There is no need to change this file

File: packages/client/src/api/api.ts (hunk 1 above)
Comment (sandheep45) [reply packages/client/src/api/api.ts:27]: See there are a lot of unnecessary data which need not to be passed to the hook. We need only `response` and `status` fields

image

Comment (prasenjeet-symon) [reply packages/client/src/api/api.ts:27]: How we will detect the network issues? To notify the client? We need the whole AxiosError ( it is not heavy on the hook believe me )

Comment (sandheep45) [reply packages/client/src/api/api.ts:27]: Yeah sure

**Pull request closed without merging**